.dotnet/src/Custom/Batch/BatchClient.Protocol.cs

```diff
@@ -10,33 +11,60 @@ public partial class BatchClient
     /// <summary>
     /// [Protocol Method] Creates and executes a batch from an uploaded file of requests
     /// </summary>
+    /// <param name="returnWhen"> <see cref="ReturnWhen.Completed"/> if the
+    /// method should return when the service has finished running the 
+    /// operation, or <see cref="ReturnWhen.Started"/> if it should return 
+    /// after the operation has been created but may not have completed 
+    /// processing. </param>
     /// <param name="content"> The content to send as the body of the request. </param>
     /// <param name="options"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
     /// <exception cref="ArgumentNullException"> <paramref name="content"/> is null. </exception>
     /// <exception cref="ClientResultException"> Service returned a non-success status code. </exception>
-    /// <returns> The response returned from the service. </returns>
-    public virtual async Task<ClientResult> CreateBatchAsync(BinaryContent content, RequestOptions options = null)
+    /// <returns> A <see cref="CreateBatchOperation"/> that can be used to wait for 
+    /// the operation to complete, or cancel the operation. </returns>
+    public virtual async Task<CreateBatchOperation> CreateBatchAsync(ReturnWhen returnWhen, BinaryContent content, RequestOptions options = null)
```

Comment:
Per offline discussion with @KrzysztofCwalina, we will remove `ReturnWhen` from SCM APIs in favor of a boolean `waitUntilCompleted` parameter where it appears currently.  This removes an unnecessary type from the SCM API and better aligns naming with the Azure.Core `WaitUntil` enum.

Reply:
Addressed.